test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/ApplicationPartManagerTest.cs

```diff
@@ -0,0 +1,148 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    public class ApplicationPartManagerTest
+    {
+        [Fact]
+        public void PopulateFeature_InvokesAllProvidersSequentially_ForAGivenFeature()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            manager.ApplicationParts.Add(new InterestPart("InterestPartA"));
+            manager.ApplicationParts.Add(new NotOfInterestPart("NotOfInterestPartB"));
+            manager.ApplicationParts.Add(new InterestPart("InterestPartC"));
+            manager.FeatureProviders.Add(
+                new InterestFeatureProvider((f, v) => f.Values.Add($"InterestFeatureProvider1{v}")));
+            manager.FeatureProviders.Add(
+                new InterestFeatureProvider((f, v) => f.Values.Add($"InterestFeatureProvider2{v}")));
+
+            var feature = new InterestFeature();
+            var expectedResults = new[] {
+                "InterestFeatureProvider1InterestPartA",
+                "InterestFeatureProvider1InterestPartC",
+                "InterestFeatureProvider2InterestPartA",
+                "InterestFeatureProvider2InterestPartC" };
+
+            // Act
+            manager.PopulateFeature(feature);
+
+            // Assert
+            Assert.Equal(expectedResults, feature.Values.ToArray());
+        }
+
+        [Fact]
+        public void PopulateFeature_InvokesOnlyProviders_ForAGivenFeature()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            manager.ApplicationParts.Add(new InterestPart("InterestPart"));
+            manager.FeatureProviders.Add(
+                new InterestFeatureProvider((f, v) => f.Values.Add($"InterestFeatureProvider{v}")));
+            manager.FeatureProviders.Add(
+                new NotInterestedFeatureProvider((f, v) => f.Values.Add($"NotOfInterestFeatureProvider{v}")));
+
+            var feature = new InterestFeature();
+            var expectedResults = new[] { "InterestPartInterestFeatureProvider" };
+
+            // Act
+            manager.PopulateFeature(feature);
+
+            // Assert
+            Assert.Equal(expectedResults, feature.Values.ToArray());
+        }
+
+        [Fact]
+        public void PopulateFeature_SkipProviders_ForOtherFeatures()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            manager.ApplicationParts.Add(new NotOfInterestPart("NotOfInterestPart"));
+            manager.FeatureProviders.Add(
+                new InterestFeatureProvider((f, v) => f.Values.Add($"InterestFeatureProvider{v}")));
+
+            var feature = new InterestFeature();
+
+            // Act
+            manager.PopulateFeature(feature);
+
+            // Assert
+            Assert.Empty(feature.Values.ToArray());
+        }
+
+        private class InterestPart : ApplicationPart
```

Comment:
Yes, it seems my earlier comment with regard to names was taken a bit too literally... perhaps still some room for improvementizationalism.